snorkel/classification/models/advanced/advanced_classifier.py

```diff
@@ -196,9 +199,7 @@ def calculate_loss(
         return loss_dict, count_dict
 
     @torch.no_grad()
-    def _calculate_probs(
-        self, X_dict: Mapping[Union[str, int], torch.Tensor], task_names: Iterable[str]
-    ):
+    def _calculate_probs(self, X_dict: Mapping[str, Any], task_names: Iterable[str]):
```

Comment:
Not all Tensors any more?

Reply:
Y will always be tensors. X doesn't have to be. e.g., you can pass string-based uids or other tags through as well that a module takes advantage of.